certs_net.py

```diff
@@ -8,61 +8,115 @@
+                self.logger.info("CertsNet: Getting secrets for all namespaces")
+                ret = v1.list_secret_for_all_namespaces()
+        except:
+            self.logger.exception("fish: retrieving kubenetes secrets failed")
+            return
+
+        tls_secrets_found = 0
+        secrets_processed = 0
         for secret in ret.items:
-            if secret.type == 'kubernetes.io/tls' and 'ca.crt' in secret.data and secret.data['ca.crt'] != '':
-                caSecondsLeft = self.getExpiry(secret.data['ca.crt'])
-                tlsSecondsLeft = self.getExpiry(secret.data['tls.crt']) 
-                self.trawler.set_gauge(
-                    'cert', 'remaining_seconds', 
-                    tlsSecondsLeft, 
-                    labels={'secret':secret.metadata.name, 'cert':'tls.crt', 'namespace': secret.metadata.namespace})
-                self.trawler.set_gauge(
-                    'cert', 'remaining_seconds', 
-                    caSecondsLeft, 
-                    labels={'secret':secret.metadata.name, 'cert':'ca.crt', 'namespace': secret.metadata.namespace})
+            if secret.type == 'kubernetes.io/tls':
+                tls_secrets_found += 1
+                this_secret_counted = False
+                if 'tls.crt' in secret.data and secret.data['tls.crt'] != '':
+                    self.logger.trace("Processing tls.crt for secret {}".format(secret.metadata.name))
+                    tlsSecondsLeft = self.getExpiry(secret.data['tls.crt'])
+                    if tlsSecondsLeft == None:
+                        continue
+                    if self.trawler:
+                        self.trawler.set_gauge(
+                            'cert', 'remaining_seconds',
+                            tlsSecondsLeft,
+                            labels={'secret':secret.metadata.name, 'cert':'tls.crt', 'namespace': secret.metadata.namespace})
+                    this_secret_counted = True
+                    secrets_processed += 1
+                if 'ca.crt' in secret.data and secret.data['ca.crt'] != '':
+                    self.logger.trace("Processing ca.crt for secret {}".format(secret.metadata.name))
+                    caSecondsLeft = self.getExpiry(secret.data['ca.crt'])
+                    if caSecondsLeft == None:
+                        continue
+                    if self.trawler:
+                        self.trawler.set_gauge(
+                            'cert', 'remaining_seconds',
+                            caSecondsLeft,
+                            labels={'secret':secret.metadata.name, 'cert':'ca.crt', 'namespace': secret.metadata.namespace})
+                    if not this_secret_counted:
+                        secrets_processed += 1
+                else:
+                    self.logger.trace("No ca.crt for secret {}".format(secret.metadata.name))
+        if self.logger:
+            self.logger.info("Finished fish. Processed {}/{}".format(secrets_processed, tls_secrets_found))
```

Comment:
Might be worth us exposing these as metrics in their own right as well so we have stats on the processing in prometheus